ddtrace/contrib/trace_utils.py

```diff
@@ -279,26 +284,48 @@ def activate_distributed_headers(tracer, int_config=None, request_headers=None,
             tracer.context_provider.activate(context)
 
 
-def flatten_dict(
-    d,  # type: Dict[str, Any]
+def _flatten(
+    obj,  # type: Any
     sep=".",  # type: str
     prefix="",  # type: str
     exclude=None,  # type: Optional[Set[str]]
 ):
-    # type: (...) -> Dict[str, Any]
-    """
-    Returns a normalized dict of depth 1
-    """
-    flat = {}
+    # type: (...) -> Generator[Tuple[str, Any], None, None]
     s = deque()  # type: ignore
-    s.append((prefix, d))
+    s.append((prefix, obj))
     exclude = exclude or set()
     while s:
         p, v = s.pop()
         if p in exclude:
             continue
         if isinstance(v, dict):
-            s.extend((p + sep + k if p else k, v) for k, v in v.items())
+            s.extend((sep.join((p, k)) if p else k, v) for k, v in v.items())
         else:
-            flat[p] = v
-    return flat
+            yield p, v
+
+
+def flatten_dict(
+    d,  # type: Dict[str, Any]
+    sep=".",  # type: str
+    prefix="",  # type: str
+    exclude=None,  # type: Optional[Set[str]]
+):
+    # type: (...) -> Dict[str, Any]
+    """
+    Returns a normalized dict of depth 1
+    """
+    return {k: v for k, v in _flatten(d, sep, prefix, exclude)}
+
+
+def set_flatten_tags(
```

Comment:
```suggestion
def set_flat_tags(
```
is maybe the grammatically correct name? 🤔

Reply:
/shrug sounds fine to me either way 😆

Reply:
`set_flattened_tags`?